### DISCUSSION TITLE
LPS node Update problem.

### DISCUSSION THREAD

**Discussion by Yuhang-Yohoo** [bitcraze/discussions#1597] [Q&A]
Hi devs, sorry to bother you,  but I haven't found helpful solutions to my problem here.

Well, the thing is that I follow all your instructions and upload 2 anchors' firmware half year ago. However, this week I got more anchors and I decide to reflash and reconfigure their ID,  but I suddenly found that all the anchors (both new and old) can not be updated by the lps-tool.

Interestingly, I can enter the dfu mode and the lps-tool can identify them, I can even click the "update" button. Unfortunately, the progress stuck and it moved to 2% after hours, I believe last time it only took me seconds to finish. I tried both on Windows and WSL in my computer but they don't work for the same problem. The ID can be changed for the new anchors but once I plug out the USB on the halfway of updating the firmware,  the node's system will be broken and since I cant flash a new firmware I can't change its ID anymore. I have to cancel the process because I waited for hours and it only went to about 30%.

Do you know what's wrong with this and what should I do? Any assistance will be appreciated.

**Comment by Yuhang-Yohoo** [comment]
Here's the picture, wait for an hour but only 2% finished
![image](https://github.com/user-attachments/assets/43182940-b28f-4d59-b270-8179f04c6768)

**Comment by Yuhang-Yohoo** [comment]
Okay, I finally fixed the problem, but still don't know why.  

Here's how to solve it : **Use a USB hub, and connect the node to the hub instead of the motherboard of the laptop.**

So this is 100% a BUG or the problem of hardware compatibility. It seems that all the USB bus, ports, and cables are working, but some of them are incompatible. So I suggest anyone have the same problem try different hardware devices and see if it can work then. 

![image](https://github.com/user-attachments/assets/92efaeb4-8e26-4fe7-b05d-b9825f41e6f2)

**Reply by gemenerik** [reply]
Glad you found a fix, and thanks for sharing it with us!
